jdbc-adapter/virtualschema-jdbc-adapter/src/main/java/com/exasol/adapter/jdbc/KerberosUtils.java

```diff
@@ -33,37 +33,28 @@ public static void configKerberos(String user, String pass) throws Exception {
         krbDir.delete();
         krbDir.mkdir();
         krbDir.deleteOnExit();
-        String krbConfPath = writeKrbConf(krbDir, confKeytab[0]);
-        String keytabPath = writeKeytab(krbDir, confKeytab[1]);
+        String krbConfPath = writePath(krbDir, confKeytab[0], "krb_", ".conf");
+        String keytabPath = writePath(krbDir, confKeytab[1], "kt_", ".keytab");
         String jaasConfigPath = writeJaasConfig(krbDir, user, keytabPath);
         System.setProperty("java.security.auth.login.config", jaasConfigPath);
         System.setProperty("java.security.krb5.conf", krbConfPath);
         System.setProperty("javax.security.auth.useSubjectCredsOnly", "false");
     }
 
-    private static String writeKrbConf(File krbDir, String base64Conf) throws Exception {
-        File file = File.createTempFile("krb_", ".conf", krbDir);
+    private static String writePath(File krbDir, String confKeyTab, String prefix, String suffix) throws Exception {
+        File file = File.createTempFile(prefix, suffix, krbDir);
         file.deleteOnExit();
-        FileOutputStream os = new FileOutputStream(file);
-        os.write(DatatypeConverter.parseBase64Binary(base64Conf));
-        os.close();
-        return file.getCanonicalPath();
-    }
-
-    private static String writeKeytab(File krbDir, String base64Keytab) throws Exception {
-        File file = File.createTempFile("kt_", ".keytab", krbDir);
-        file.deleteOnExit();
-        FileOutputStream os = new FileOutputStream(file);
-        os.write(DatatypeConverter.parseBase64Binary(base64Keytab));
-        os.close();
+        try (FileOutputStream os = new FileOutputStream(file);) {
+            os.write(DatatypeConverter.parseBase64Binary(confKeyTab));
+        }
         return file.getCanonicalPath();
     }
 
     private static String writeJaasConfig(File krbDir, String princ, String keytabPath) throws Exception {
         File file = File.createTempFile("jaas_", ".conf", krbDir);
         file.deleteOnExit();
         String jaasData;
```

Comment:
Use `StringBuilder`;

Reply:
Can't do it safely without an unit test. Should be done in a separated branch.